python/torch_mlir_e2e_test/test_suite/indirect_memory.py

```diff
@@ -0,0 +1,158 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+import torch
+
+from torch_mlir_e2e_test.framework import TestUtils
+from torch_mlir_e2e_test.registry import register_test_case
+from torch_mlir_e2e_test.annotations import annotate_args, export
+
+# ==============================================================================
+
+class ScatterReduceFloatModule(torch.nn.Module):
+    include_self: bool
+    reduce_type: str
+
+    def __init__(self, reduce_type: str, include_self: bool):
+        super().__init__()
+        self.include_self = include_self
+        self.reduce_type = reduce_type
+
+    @export
+    @annotate_args([
+        None,
+        ([10, 8, 6], torch.float32, True),
+        ([2, 4, 3], torch.int64, True),
+        ([5, 8, 6], torch.float32, True),
```

Comment:
Does the code not work with dynamic shapes?

Reply:
Switched the annotations to make them dynamic